core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcRules.java

```diff
@@ -332,6 +332,9 @@ protected JdbcJoinRule(Config config) {
      * @return Whether condition is supported
      */
     private static boolean canJoinOnCondition(RexNode node) {
+      if (node.isAlwaysTrue() || node.isAlwaysFalse()) {
+        return true;
+      }
```

Comment:
Add another case condition:
```
// ***************
case LITERAL:
        return true;
```
is more suitable ?

Reply:
i agree. with a comment that this would be TRUE or FALSE.

Reply:
shouldn't I check that then the literal is in fact a boolean?
```
case LITERAL:
        return ((RexLiteral) node).getTypeName() == SqlTypeName.BOOLEAN;
```

or perhaps there is some guarantee that something that is not a boolean can't show up there?

Reply:
`Join#isValid` ensures that the condition, in this case the literal, is always a Boolean so there is no need for additional checks thus marking this conversation as resolved.